### PR TITLE
(DOCSP-28190) Implement Andrew's RN SDK docs feedback - Round 2

### PR DESCRIPTION
**Pull request by krollins-mdb**
## Pull Request Info

Covers items 2 - 4 in the Jira ticket.

### Jira

- https://jira.mongodb.org/browse/DOCSP-28190

### Staged Changes

- [CRUD - Read](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-28190-round2/sdk/react-native/realm-database/crud/read/)
- [CRUD - Update](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-28190-round2/sdk/react-native/realm-database/crud/update/)
- [Schema - Default value](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-28190-round2/sdk/react-native/realm-database/schemas/default-value/)
- [Schema - Optional property](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-28190-round2/sdk/react-native/realm-database/schemas/optional/)

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: examples/react-native/__tests__/js/Models/Car.js [hunk 1]
```diff
@@ -8,6 +8,10 @@ class Car extends Realm.Object {
       make: 'string',
       model: 'string',
       miles: {type: 'int', default: 0},
+      timestamp: {
+        type: 'int',
+        default: () => Math.round(new Date().getTime() / 1000),
```
Comment (mongodben) [inline examples/react-native/__tests__/js/Models/Car.js:13]: i believe default doesn't take a function but rather just the value

File: source/examples/generated/react-native/js/Car.snippet.js-car-schema.js [hunk 2]
```diff
@@ -5,6 +5,10 @@ class Car extends Realm.Object {
       make: 'string',
       model: 'string',
       miles: {type: 'int', default: 0},
+      timestamp: {
+        type: 'int',
+        default: () => Math.round(new Date().getTime() / 1000),
```
Comment (mongodben) [inline source/examples/generated/react-native/js/Car.snippet.js-car-schema.js:10]: ditto https://github.com/mongodb/docs-realm/pull/2696/files#r1151097001

File: source/sdk/react-native/realm-database/crud/read.txt [hunk 3]
```diff
@@ -41,8 +41,11 @@ Read Operations
 ---------------
 
 Read operations are queries to find your data stored in Realm Database. Use the
-``@realm/react`` hooks to find specific objects and get collections by object
-type.
+``@realm/react`` hooks ``useObject()`` and ``useQuery()`` to find specific objects 
+and get collections by object type.
```
Comment (mongodben) [inline source/sdk/react-native/realm-database/crud/read.txt:45]:
```suggestion
following ``@realm/react`` hooks to read data in a realm:

- ``useObject()``: Find a specific object by primary key.
- ``useQuery()``: Get a collection of objects by object type.
```

File: source/sdk/react-native/realm-database/crud/read.txt [hunk 4]
```diff
@@ -41,8 +41,11 @@ Read Operations
...
+
+These hooks return live objects, which are automatically updated. When an object
```
Comment (mongodben) [inline source/sdk/react-native/realm-database/crud/read.txt:47]:
```suggestion
These hooks return live objects, which are automatically updated when the data in the realm changes.
```

File: source/sdk/react-native/realm-database/crud/read.txt [hunk 5]
```diff
@@ -41,8 +41,11 @@ Read Operations
...
+returned by a hook is updated, the parent component is re-rendered.
```
Comment (mongodben) [inline source/sdk/react-native/realm-database/crud/read.txt:48]:
```suggestion
These hooks return live objects, which are automatically updated when the data in the realm changes. When objects returned by these hooks are updated,
the component calling the hook rerenders.
```

File: source/sdk/react-native/realm-database/schemas/default-value.txt [hunk 6]
```diff
@@ -22,7 +22,12 @@ Example
 ~~~~~~~
 
 In the following example of a ``Car`` class, we define a ``miles`` property with
-a default value of 0.
+a default value of ``0``. 
+
+.. versionadded:: 11.1.0
+
+In Realm.js v11.1.0 and later, you can use a function to define a dynamic
+default value, like the ``timestamp`` property in the example below.
```
Comment (mongodben) [inline source/sdk/react-native/realm-database/schemas/default-value.txt:30]: oh TIL!

Comment (krollins-mdb) [reply source/sdk/react-native/realm-database/schemas/default-value.txt:30]: I didn't know either until I read Andrew's initial comment. Lots of useful info from Andrew!

File: source/sdk/react-native/realm-database/schemas/optional.txt [hunk 7]
```diff
@@ -20,7 +20,8 @@ To mark a property as optional, append a question mark ``?`` to its type.
 Example
```
Comment (takameyer) [inline source/sdk/react-native/realm-database/schemas/optional.txt:20]: Can we extend the overview text above to indicate that it's possible to use the `optional` boolean?

**Review by takameyer — APPROVED** [review]
Just one suggestion.  Otherwise, LGTM :shipit:

File: source/sdk/react-native/realm-database/schemas/optional.txt (hunk 7 above)
Comment (krollins-mdb) [reply source/sdk/react-native/realm-database/schemas/optional.txt:20]: I've updated to make this clearer. Thanks, Andrew!

**Pull request merged**